server/website/script/fixture_generators/knob_settings/oracle/create_knob_settings.py

```diff
@@ -0,0 +1,210 @@
+    if fields['name'].upper() == 'LARGE_POOL_SIZE':
+        fields['tunable'] = True
+        fields['minval'] = 0
+        fields['maxval'] = 2000000000  # 2GB
+        fields['default'] = 500000000  # 500M
+    if fields['name'].upper() == 'PGA_AGGREGATE_TARGET':
+        fields['tunable'] = False
+        fields['minval'] = 0
+        fields['maxval'] = 33000000000  # 33G
+        fields['default'] = 0
+    if fields['name'].lower() == 'bitmap_merge_area_size':
+        fields['tunable'] = True
+        fields['minval'] = 0
+        fields['maxval'] = 5000000000  # 3G
+        fields['default'] = 0
+    if fields['name'].lower() == 'create_bitmap_area_size':
+        fields['tunable'] = True
+        fields['minval'] = 0
+        fields['maxval'] = 5000000000  # 3G
+        fields['default'] = 0
+    if fields['name'].lower() == 'hash_area_size':
+        fields['tunable'] = True
+        fields['minval'] = 0
+        fields['maxval'] = 3000000000  # 3G
+        fields['default'] = 0
+    if fields['name'].lower() == 'sort_area_size':
+        fields['tunable'] = True
+        fields['minval'] = 0
+        fields['maxval'] = 3000000000  # 3G
+        fields['default'] = 0
+    if fields['name'].upper() == 'OPEN_CURSORS':
+        fields['tunable'] = False
+        fields['minval'] = 200
+        fields['maxval'] = 400
+        fields['default'] = 300
+    if fields['name'].upper() == 'DB_FILE_MULTIBLOCK_READ_COUNT':
+        fields['tunable'] = False
+        fields['minval'] = 64
+        fields['maxval'] = 256
+        fields['default'] = 128
+    if fields['name'].upper() == 'optimizer_index_cost_adj'.upper():
+        fields['tunable'] = False
+        fields['minval'] = 1
+        fields['maxval'] = 10000
+        fields['default'] = 100
+    if fields['name'].upper() == 'OPTIMIZER_USE_PENDING_STATISTICS':
+        fields['tunable'] = False
+        fields['minval'] = None
+        fields['maxval'] = None
+        fields['default'] = False
+    if fields['name'].upper() == 'OPTIMIZER_USE_INVISIBLE_INDEXES':
+        fields['tunable'] = False
+        fields['minval'] = None
+        fields['maxval'] = None
+        fields['default'] = False
+    if fields['name'].upper() == 'OPTIMIZER_USE_SQL_PLAN_BASELINES':
+        fields['tunable'] = False
+        fields['minval'] = None
+        fields['maxval'] = None
+        fields['default'] = True
+    if fields['name'].upper() == 'OPTIMIZER_CAPTURE_SQL_PLAN_BASELINES':
+        fields['tunable'] = False
+        fields['minval'] = None
+        fields['maxval'] = None
+        fields['default'] = False
+    if fields['name'].upper() == 'DISK_ASYNCH_IO':
+        fields['tunable'] = True
+        fields['vartype'] = 5
+        fields['enumvals'] = 'TRUE,FALSE'
+        fields['default'] = 'TRUE'
+
+
+def main():
+    final_metrics = []
+    with open('oracle.txt', 'r') as f:
+        num = 0
+
+        lines = f.readlines()
+        for line in lines:
```

Comment:
Hum... got it what you did here. Okay, I'd suggest you treat the entries in the list before traversing it, as below:
```
lines = f.readlines()
lines = [i.strip('\n').strip() for i in lines]
lines = [i for i in lines if i and i not in ['NAME','TYPE','DESCRIPTION'] and not i.startswith('-')] 
      ...
```
You could broke into more lines for readability (see that I check three conditions in the second for comprehension):
* i is not empty
* i doesn't start with '-'
* i is not DESCRIPTION, NAME and TYPE

Another possible improvement (test with care) would be:
```
for idx, line in enumerate(line):
    num = idx % 3

    if num == 0:
     ...
```
No need to declare num previously now. 

This replaces lines 166 - 170